lucupy/minimodel/program.py

```diff
@@ -143,7 +143,7 @@ def get_group(self, group_id: GroupID) -> Optional[Group]:
         def aux(group: Group) -> Optional[Group]:
             if group.id == group_id:
                 return group
-            elif group.is_scheduling_group():
+            elif not isinstance(group.children, Observation):
```

Comment:
It's too bad that mypy requires this immediate checking to confirm types, but still impressive that it can do it.